javascript/ql/src/experimental/Security/CWE-614/InsecureCookie.qll

```diff
@@ -0,0 +1,161 @@
+/**
+ * Provides classes for reasoning about cookies added to response without the 'secure' flag being set.
+ * A cookie without the 'secure' flag being set can be intercepted and read by a malicious user.
+ */
+
+import javascript
+
+module InsecureCookie {
+  /**
+   * `secure` property of the cookie options.
+   */
+  string flag() { result = "secure" }
+
+  /**
+   * Abstract class to represent different cases of insecure cookie settings.
+   */
+  abstract class InsecureCookies extends DataFlow::Node {
+    /**
+     * The name of the middleware/library used to set the cookie.
+     */
+    abstract string getKind();
+
+    /**
+     * The `cookie` options.
+     */
+    abstract DataFlow::Node getCookieOptionsArgument();
+
+    /**
+     * Predicate that determines if a cookie is insecure.
+     */
+    abstract predicate isInsecure();
```

Comment:
```suggestion
     * Holds if this cookie is secure.
     */
    abstract predicate isSecure();
```
Optional: I think it is clearer to flip the polarity and avoid the negation in the predicate name.
Please note that this requires a change to all implementations of this predicate below.

Reply:
Yes, I agree with you. I changed the polarity (I did it in one commit) of all the implementations of this predicate.